google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectTest.java

```diff
@@ -1328,7 +1675,7 @@ private void testAutoContentType(String method) throws IOException {
     String[] names = {"file1.txt", "dir with spaces/Pic.Jpg", "no_extension"};
```

Comment:
These tests are currently failing, it is unclear to me if we need to be concerned with having content type default to "application/octet-stream" for GRPC like it is in JSON. if not I can separate the logic and turn off the ones checking the default value.

Reply:
Both JSON and grpc are using the same mechanism to automatically determine content-type based on extension. I'll try to dig in an see what's going on.

Reply:
I've just posted a PR with some tests for some of the common file extensions to validate our handling https://github.com/googleapis/java-storage/pull/1647